src/net_processing.cpp

```diff
@@ -2295,7 +2295,9 @@ bool PeerManagerImpl::AlreadyHaveTx(const GenTxid& gtxid, bool include_reconside
 
     const uint256& hash = gtxid.GetHash();
 
-    if (m_orphanage.HaveTx(gtxid)) return true;
+    // Orphanage is checked by wtxid. However, even if this is a txid, look up the same hash in
```

Comment:
imo `wtxid` and `txid` being double SHA256 outputs and wtxid being essentially a superset of txid, makes this blind conversion legitimate.